### DISCUSSION TITLE
Compilation of TomoATT v1.0.0 with SIMD enabled on Aarch64 machine

### DISCUSSION THREAD

**Discussion by JingChen-Thu** [TomoATT/TomoATT#4] [Q&A] [answered]
One of our user met a problem when compiling v1.0.0 with SIMD enabled:

[logfile.log](https://github.com/user-attachments/files/17856656/logfile.log)

Commenting Line 101 in the CMakelists.txt can address this problem. @mnagaso Could you please check the compilation with SIMD? By the way, in my previous test, enabling SIMD may significantly increase the memory requirement for those processors which do not store parameters but only calculate equations on grids by using shared memory (level 3 parallelization).

**Comment by mnagaso** [comment]
@JingChen-Thu I got it!
Can I also have the log from `cmake` command as well?

**Reply by mnagaso** [reply]
On my side, the compilation with SIMD has no problem.
Yes SIMD optionzation requires more memory for preparing additional arrays, which are used for storing the values in the accessing order in each iteration.

**Comment by JingChen-Thu** [comment]
Thank you, Masaru. Here is the log of cmake from the user. He mentioned that '-mfma' was deleted since the code was compiled on aarch64 CPU.

[cmake.log](https://github.com/user-attachments/files/17870615/cmake.log)

I will also test it later.

**Comment by mnagaso** [comment, marked as answer]
Thanks! If the machine is aarch64, it may be necessary to use clang compilers for making SIMD avaialble.
Would it be possible to ask her/him to use clang compilers, which may be spacified with `CC=(Clang C compiler) CXX=(Clang C++ compiler) cmake ..`
